src/os/bluestore/NVMEDevice.cc

```diff
@@ -98,9 +97,8 @@ class SharedDriverData {
         trid(trid_),
         ctrlr(c),
         ns(ns_) {
-    sector_size = spdk_nvme_ns_get_sector_size(ns);
-    block_size = std::max(CEPH_PAGE_SIZE, sector_size);
-    size = ((uint64_t)sector_size) * spdk_nvme_ns_get_num_sectors(ns);
+    block_size = spdk_nvme_ns_get_extended_sector_size(ns);
+    size = block_size * spdk_nvme_ns_get_num_sectors(ns);
```

Comment:
@tone-zhang sorry for the confusion. i don't think we can calc the size using `block_size * spdk_nvme_ns_get_num_sectors(ns);`, instead, we can either use

```c
size = spdk_nvme_ns_get_size(ns)
```
or 
```c
size = sector_size * spdk_nvme_ns_get_num_sectors(ns);
```

as block size is the unit for addressing, which takes the overhead (metadata) into consideration, while the "size" here is the total size we can use for as data storage.

Reply:
Thanks. I think spdk_nvme_ns_get_size(ns) is better. I will update the code.